doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -65,33 +65,32 @@ The diagram::
             |-----------------------------------------> Dec19
                       |-----------------------------------------> Nov20
 
-shows the 42 month support windows for Python.  A project with a
-major or minor version release in Feb19 should support py35 and newer,
-a project with a major or minor version release in Dec19 should
-support py36 and newer, and a project with a major or minor version
-release in Nov20 should support py37 and newer.
+It shows the 42 month support windows for Python.  A project with a
+major or minor version release in February 2019 should support Python 3.5 and newer,
+a project with a major or minor version released in December 2019 should
+support Python 3.6 and newer, and a project with a major or minor version
+release in November 2020 should support Python 3.7 and newer.
 
 The current Python release cadence is 18 months so a 42 month window
 ensures that there will always be at least two minor versions of Python
-in the window.  By padding the window by 6 months from the anticipated
-Python cadence we avoid the edge cases where a project releases
-the month after Python and would effectively only support one
-minor version of Python that has an installed base.
-This six month buffer provides resilience to minor fluctuations /
-delays in the Python release schedule.
-
-Because Python minor version support is based on historical release
-dates, a 36 month time window, and a project's plans, a project can
-decide to drop a given minor version of Python very early in the release
-process.
-
-While there will be some unavoidable mismatch in supported versions of
-Python between projects if releases occurs immediately after a
-minor version of Python ages out.  This should not last longer than one
-release cycle of each of the projects, and when a given project does a
-minor or major release, it is guaranteed that there will be a stable
-release of all other projects that support the set of Python the
-new release will support.
+in the window.  The window is extended 6 months beyond the anticipated two-release
+interval for Python to provides resilience against small fluctuations /
+delays in its release schedule.
+
+Because Python minor version support is based only on historical
+release dates, a 42 month time window, and a planned project release
+date, one can predict with high confidence when a project will be able
+to drop any given minor version of Python.  This, in turn, could save
+months of unnecessary maintenance burden.
+
+If a project releases immediately after a minor version of Python
```

Comment:
@tacaswell I am now happy with the wording of the whole NEP, except for these two paragraphs, which I am still not so sure about.  It seems to me as though there are *two* situations that create discrepancies: a release shortly after a minor version drops out of support, or shortly after a new minor version enters.  In either case, the project will support a different subsets of Python.  Subsequent releases of those packages will fix it up, but I think the word "guaranteed" below is then too strong a statement.  We cannot guarantee when packages release, so if they take a long time to release after a new version of Python is introduced we're in trouble anyway.

Reply:
In the guarantee for releases right after a python minor enters support I am counting on the back-compatibility between minor versions of Python such that versions of our projects that work on 3.N, should work out of the box with version 3.(N+1). 

This back-compatibility has not been perfect on the c-api front, but a combination of cython + cpython seems to be pretty back-compatible.

Also related, everyone needs to stop shipping cythonized .c files in sdists, but that is a different discussion.

We could hedge this with  "conditional on Python's back-compat" or change it to "we expect that there will be a non-empty set of supported Python versions by all actively maintained projects"

Reply:
OK, I see where the confusion came in: I was thinking of *binary* distributions, and you are talking about source.  The source will be compatible, while the *binary* support may not be available.  The *binary* support depends on the team releasing rapidly, while the *source* support is there without any action on the developers' behalf.

With skimage, we ship Cythonized files that are used *only* when Cython is not available; problematic?

W.r.t. the "hedging", if we are talking about source that does not seem necessary; Python should not break API compatibility on minor versions, and the latter suggested phrase feels so pessimistic compared to the likely reality.

Reply:
Resolve by ee9ba282886ed1c3a927abcfe21cf42242f5d579.